File: day_one/reduce_to_zero.py
# My answer
class Solution:
    def numberOfSteps (self, num: int) -> int:
        count = 0
        while num != 0:
            if num % 2 == 0:
                num = num / 2
                count = count + 1
            else:
                num = num - 1
                count = count + 1
        return count

# Cleanest answer
class Solution:
    def numberOfSteps (self, num: int) -> int:
        digits = f'{num:b}'
        return digits.count('1') - 1 + len(digits)

File: day_one/reduce_to_zero.js
/**
 * @param {number} num
 * @return {number}
 */
var numberOfSteps  = function(num) {
    let answer = 0
    while (num !== 0){
        if (num % 2 === 0){
            num = num / 2
            answer = answer + 1
        }
        else if (num > 0){
            num = num -1
            answer = answer + 1
        }
    }
    return answer
};
